test/test_gem_compatibility.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+# JRuby does not support forking, and it doesn't seem worth the effort to make it work.
+return unless Process.respond_to?(:fork)
+
+require_relative 'helper'
+
+describe 'gem compatibility' do
+  %w[
+    resolv-replace
+  ].each do |gem_name|
+    it "passes smoke test with #{gem_name.inspect} gem required" do
+      memcached(:binary, rand(21_397..21_896)) do |_, port|
+        in_isolation(timeout: 10) do
```

Comment:
We want to be able to require the gem in isolation so as not to affect the host process and other tests.